vignettes/target_types.Rmd

```diff
@@ -0,0 +1,49 @@
+---
+title: "Target type guidelines"
+author: "David Watkins"
+date: "3/25/2019"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Target types}
+  %\VignetteEngine{knitr::rmarkdown}
+  \usepackage[utf8]{inputenc}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+`scipiper` targets can be indicator files with a [shared cache](shared_cache.html), local data files with or without local indicator files, or simply R objects.  This vignette provides some guidelines for when to use these different patterns.
+
+## Objectives
+There are two main goals to keep in mind when deciding what style of target to use:
+ 
+ - **Minimize time spend unnecessarily rebuilding targets** --- i.e., those that have already been built locally or by collaborators
+ - **Keeping the pipeline simple** --- caching can save lots of build time, but also adds to the code that has to be maintained, can clutter the repo with build and indicator files, etc.
+ 
+ These two can sometimes be in tension, but usually it will be clear which is more important for a given case.
+ 
+## Key factors to consider
+
+ - **Time to create an object** vs **object size**: This is the biggest thing to consider --- is it faster to upload a file to the shared cache and have others download it, or just have others rebuild it themselves?  This will somewhat depend on your internet connection.  Don't forget about upload time as well.  Uploading is generally slower than downloading, especially for remote workers with residential internet connections.   
```

Comment:
I'm skeptical that our residential internet connections are noticeably slower than the Middleton connection... :stuck_out_tongue_winking_eye:

Reply:
Note that downloading is sometimes optional for many teammates, such that the file may only need to be present on one person's computer and the shared cache.

Reply:
The title to this bullet is about time vs size, but the bullet text focuses on size only as it relates to internet transfer speed. That's an important consideration, but if we're not also concerned about storing large files on a shared cache because of the storage space/cost itself, maybe the title should call out transfer time rather than object size.

Reply:
Re: speeds -- my downstream is definitely faster than the office, but my upload may actually be worse.  Residential internet usually has pretty heavily throttled upload speeds, whereas business connections are more equal upstream/downstream.